### PR TITLE
Code Quality: enable a few useful ESLint rules

### PR DESCRIPTION
**Pull request by swissspidy**
## Context

Related to #12086

## Summary

While the solutions mentioned on that ticket are a bit outdated and not super useful for us, I was able to steal a couple of useful ESLint rules from them for our project.

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -35,6 +36,8 @@
     "consistent-return": "error",
     "curly": ["error", "all"],
     "default-case": "error",
+    "eol-last": "error",
```
Comment (miina) [inline .eslintrc:39]: **💡 thought:**
Should we also use [no-multiple-empty-lines](https://eslint.org/docs/latest/rules/no-multiple-empty-lines) together with it? To enforce just one empty line.

File: packages/story-editor/src/components/panels/design/textStyle/test/textStyle.js [hunk 2]
```diff
@@ -63,6 +64,7 @@ jest.mock('../../../../form/color/color', () => {
   const React = require('@googleforcreators/react');
 
   const _PropTypes = require('prop-types');
+  // eslint-disable-next-line prefer-arrow-callback -- False positive.
   const FakeControl = React.forwardRef(function FakeControl(props, ref) {
```
Comment (miina) [inline packages/story-editor/src/components/panels/design/textStyle/test/textStyle.js:68]: **❓question:**
Curious, doesn't it work the same way here as elsewhere? (`forwardRef((props, ref) => {...})`)

**Review by miina — COMMENTED** [review]
Looks like unit tests are failing now on this PR.

File: packages/story-editor/src/components/panels/design/textStyle/test/textStyle.js (hunk 2 above)
Comment (swissspidy) [reply packages/story-editor/src/components/panels/design/textStyle/test/textStyle.js:68]: Then there will be a lint warning about a missing component display name.

File: .eslintrc (hunk 1 above)
Comment (swissspidy) [reply .eslintrc:39]: Good idea, implemented.

**Review by miina — APPROVED** [review]

**Review by timarney — APPROVED** [review]

File: packages/story-editor/src/components/panels/design/textStyle/test/textStyle.js (hunk 2 above)
Comment (swissspidy) [reply packages/story-editor/src/components/panels/design/textStyle/test/textStyle.js:68]: I see now what you meant and fixed the ESLint config & undid the other related changes.

**Review by merapi — APPROVED** [review]

**Pull request merged**